vm_test.go

```diff
@@ -229,6 +230,7 @@ func BenchmarkEmptyLoop(b *testing.B) {
 	`
 	b.StopTimer()
 	vm := New()
+	vm.SetRateLimiter(rate.NewLimiter(rate.Limit(10_000_000), 250_000))
```

Comment:
[opt/q] Thoughts on setting a default rate limiter within `(r *Runtime) init()` instead? I don't really have a strong preference, but I just wanted to ask.

Reply:
Yeah that's probably the best way to do it so we don't have to update each test.

Reply:
Just giving myself a reminder to make sure adding this to init doesn't introduce a significant slowdown.

Reply:
with default rate limiter
```
goja (BAAS-23656_removeSelfLimiterNilCheck)$ go test -bench=^BenchmarkVMInitCost$ -run=^$ -benchmem -benchtime 1000000x
goos: darwin
goarch: arm64
pkg: github.com/dop251/goja
BenchmarkVMInitCost-10           1000000             76075 ns/op          157266 B/op       1825 allocs/op
PASS
ok      github.com/dop251/goja  76.269s
```

without default rate limiter
```
goja (BAAS-23656_removeSelfLimiterNilCheck)$ go test -bench=^BenchmarkVMInitCost$ -run=^$ -benchmem -benchtime 1000000x
goos: darwin
goarch: arm64
pkg: github.com/dop251/goja
BenchmarkVMInitCost-10           1000000             77422 ns/op          157187 B/op       1824 allocs/op
PASS
ok      github.com/dop251/goja  77.632s
```

Differences were negligible, it is definitely worth including this.